src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -13,35 +13,54 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the task identified by the index number used in the last task listing.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
+            + ": Deletes the task identified by its name or the index number used in the last task listing.\n"
+            + "Parameters: INDEX (must be a positive integer) or NAME (the name of a task to delete)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Task: %1$s";
 
     public final int targetIndex;
+    public final String targetName;
 
     public DeleteCommand(int targetIndex) {
         this.targetIndex = targetIndex;
+        this.targetName = null;
```

Comment:
Both are final variables (following the design of the original source code) thus upon first initialization you cant change its value any more. And depending on the constructor (we have two now) one has to be set to a valid value. Thats why its not initialized upon definition. If you were to read original source they did the same thing except with only the targetIndex variable. It is self explanatory from seeing the diff.